### PR TITLE
V14 QA content image media picker tests

### PR DESCRIPTION
**Pull request by andr317c**
Added tests for the Image media picker in content

### CODE REVIEW

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+  await umbracoApi.document.ensureNameNotExists(contentName);
+  await umbracoApi.documentType.ensureNameNotExists(documentTypeName);
+});
+
+test('can save content with a image media picker', async ({umbracoApi, umbracoUi}) => {
+  // Arrange
+  const expectedState = 'Draft';
+  const dataType = await umbracoApi.dataType.getByName(dataTypeName);
+  await umbracoApi.documentType.createDocumentTypeWithPropertyEditor(documentTypeName, dataTypeName, dataType.id, groupName);
+  await umbracoUi.goToBackOffice();
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts:25]: This should be in the "beforeEach"

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+  const dataTypeId = await umbracoApi.dataType.createImageMediaPickerDataType(customDataTypeName, 0,1, true);
+  const documentId = await umbracoApi.documentType.createDocumentTypeWithPropertyEditor(documentTypeName, customDataTypeName, dataTypeId, groupName);
+  await umbracoApi.document.createDocumentWithImageMediaPicker(contentName, documentId, AliasHelper.toAlias(customDataTypeName), imageId, {left: 0.4, top: 0.2});
+  await umbracoUi.goToBackOffice();
+  await umbracoUi.content.goToSection(ConstantHelper.sections.content);
+
+  // Act
+  await umbracoUi.content.goToContentWithName(contentName);
+  await umbracoUi.content.clickExactLinkWithName(mediaName);
+  await umbracoUi.content.clickResetFocalPointButton()
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts:220]:
```suggestion
  await umbracoUi.content.clickResetFocalPointButton();
```

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+  await umbracoUi.content.isSuccessNotificationVisible();
+  expect(await umbracoApi.document.doesImageMediaPickerContainImageWithFocalPoint(contentName, AliasHelper.toAlias(customDataTypeName), imageId, {left: 0.4, top: 0.2})).toBeTruthy();
+});
+
+test('can reset focal point in a image from the image media picker', async ({umbracoApi, umbracoUi}) => {
+  // Arrange
+  await umbracoApi.media.ensureNameNotExists(mediaName);
+  await umbracoApi.dataType.ensureNameNotExists(customDataTypeName);
+  const imageId = await umbracoApi.media.createDefaultMediaWithImage(mediaName);
+  const dataTypeId = await umbracoApi.dataType.createImageMediaPickerDataType(customDataTypeName, 0,1, true);
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts:211]:
```suggestion
  const dataTypeId = await umbracoApi.dataType.createImageMediaPickerDataType(customDataTypeName, 0, 1, true);
```

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+
+  // Clean
+  await umbracoApi.dataType.ensureNameNotExists(customDataTypeName);
+});
+
+// I get no error notification
+test.skip('image count can not be more that max amount set in image media picker', async ({umbracoApi, umbracoUi}) => {
+  // Arrange
+  await umbracoApi.dataType.ensureNameNotExists(customDataTypeName);
+  const dataTypeId = await umbracoApi.dataType.createImageMediaPickerDataType(customDataTypeName, 0,0);
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts:135]:
```suggestion
  const dataTypeId = await umbracoApi.dataType.createImageMediaPickerDataType(customDataTypeName, 0, 0);
```

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+  await umbracoApi.media.ensureNameNotExists(mediaFolderName);
+  await umbracoApi.dataType.ensureNameNotExists(customDataTypeName);
+});
+
+test('can add an image from the image media picker with focal point enabled', async ({umbracoApi, umbracoUi}) => {
+  // Arrange
+  await umbracoApi.media.ensureNameNotExists(mediaName);
+  await umbracoApi.dataType.ensureNameNotExists(customDataTypeName);
+  const imageId = await umbracoApi.media.createDefaultMediaWithImage(mediaName);
+  const dataTypeId = await umbracoApi.dataType.createImageMediaPickerDataType(customDataTypeName, 0,1, true);
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts:188]:
```suggestion
  const dataTypeId = await umbracoApi.dataType.createImageMediaPickerDataType(customDataTypeName, 0, 1, true);
```

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+  await umbracoApi.media.ensureNameNotExists(mediaName);
+});
+
+test('can remove an image from the image media picker', async ({umbracoApi, umbracoUi}) => {
+  // Arrange
+  const dataType = await umbracoApi.dataType.getByName(dataTypeName);
+  await umbracoApi.media.ensureNameNotExists(mediaName);
+  const imageId = await umbracoApi.media.createDefaultMediaWithImage(mediaName);
+  const documentId = await umbracoApi.documentType.createDocumentTypeWithPropertyEditor(documentTypeName, dataTypeName, dataType.id, groupName);
+  await umbracoApi.document.createDocumentWithImageMediaPicker(contentName, documentId, AliasHelper.toAlias(dataTypeName), imageId);
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts:92]: AliasHelper.toAlias(dataTypeName) should be defined as a variable.

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+});
+
+test('can add an image from the image media picker with focal point enabled', async ({umbracoApi, umbracoUi}) => {
+  // Arrange
+  await umbracoApi.media.ensureNameNotExists(mediaName);
+  await umbracoApi.dataType.ensureNameNotExists(customDataTypeName);
+  const imageId = await umbracoApi.media.createDefaultMediaWithImage(mediaName);
+  const dataTypeId = await umbracoApi.dataType.createImageMediaPickerDataType(customDataTypeName, 0,1, true);
+  const documentId = await umbracoApi.documentType.createDocumentTypeWithPropertyEditor(documentTypeName, customDataTypeName, dataTypeId, groupName);
+  await umbracoApi.document.createDocumentWithImageMediaPicker(contentName, documentId, AliasHelper.toAlias(customDataTypeName), imageId);
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts:190]: AliasHelper.toAlias(customDataTypeName) should be defined as a variable.

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+  await umbracoApi.dataType.ensureNameNotExists(customDataTypeName);
+  const imageId = await umbracoApi.media.createDefaultMediaWithImage(mediaName);
+  const dataTypeId = await umbracoApi.dataType.createImageMediaPickerDataTypeWithCrop(customDataTypeName, cropLabel, cropWidth, cropHeight);
+  const documentId = await umbracoApi.documentType.createDocumentTypeWithPropertyEditor(documentTypeName, customDataTypeName, dataTypeId, groupName);
+  await umbracoApi.document.createDocumentWithImageMediaPicker(contentName, documentId, AliasHelper.toAlias(customDataTypeName), imageId);
+  await umbracoUi.goToBackOffice();
+  await umbracoUi.content.goToSection(ConstantHelper.sections.content);
+
+  // Act
+  await umbracoUi.content.goToContentWithName(contentName);
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts:245]: Added TODO here

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+  await umbracoUi.content.clickResetFocalPointButton()
+  await umbracoUi.content.clickSubmitButton();
+  await umbracoUi.content.clickSaveButton();
+
+  // Assert
+  await umbracoUi.content.isSuccessNotificationVisible();
+  expect(await umbracoApi.document.doesImageMediaPickerContainImageWithFocalPoint(contentName, AliasHelper.toAlias(customDataTypeName), imageId, {left: 0, top: 0})).toBeTruthy();
+});
+
+// The crop is not being selected
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts:229]:
```suggestion
// TODO: Remove skip when the front-end is ready as currently the crop is not being selected.
```

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+  await umbracoUi.content.clickSaveButton();
+
+  // Assert
+  await umbracoUi.content.isErrorNotificationVisible();
+
+  // Clean
+  await umbracoApi.dataType.ensureNameNotExists(customDataTypeName);
+});
+
+// I get no error notification
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts:131]: // TODO:  Remove skip when the front-end is ready as there are currently no displayed error notification.

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+
+  // Assert
+  await umbracoUi.content.isSuccessNotificationVisible();
+  expect(await umbracoApi.document.doesImageMediaPickerContainImage(contentName, AliasHelper.toAlias(dataTypeName), imageId)).toBeFalsy();
+
+  // Clean
+  await umbracoApi.media.ensureNameNotExists(mediaName);
+});
+
+// I get no error notification
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts:110]: //  TODO: Remove skip when the front-end is ready as there are currently no displayed error notification.

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts [hunk 12]
```diff
@@ -0,0 +1,246 @@
...
+  const dataTypeId = await umbracoApi.dataType.createImageMediaPickerDataType(customDataTypeName, 0,1, true);
+  const documentId = await umbracoApi.documentType.createDocumentTypeWithPropertyEditor(documentTypeName, customDataTypeName, dataTypeId, groupName);
+  await umbracoApi.document.createDocumentWithImageMediaPicker(contentName, documentId, AliasHelper.toAlias(customDataTypeName), imageId);
+  await umbracoUi.goToBackOffice();
+  await umbracoUi.content.goToSection(ConstantHelper.sections.content);
+
+  // Act
+  await umbracoUi.content.goToContentWithName(contentName);
+  await umbracoUi.content.clickExactLinkWithName(mediaName);
+  await umbracoUi.content.setFocalPoint(40, 20);
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts:197]: Assigning the width percentage and height percentage to variables can improve the clarity of the code.

**Review by nhudinh0309 — CHANGES_REQUESTED** [review]
I had some comments

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts (hunk 1 above)
Comment (andr317c) [reply tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts:25]: This is not possible as we are creating a customDataType in some tests

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts [hunk 13]
```diff
@@ -0,0 +1,257 @@
...
+import {expect} from "@playwright/test";
+
+const contentName = 'TestContent';
+const documentTypeName = 'TestDocumentTypeForContent';
+const dataTypeName = 'Image Media Picker';
+const customDataTypeName = 'Custom Image Media Picker';
+const groupName = 'TestGroup';
+const mediaName = 'TestImage';
+
+test.beforeEach(async ({umbracoApi, umbracoUi}) => {
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts:11]:
```suggestion
test.beforeEach(async ({umbracoApi}) => {
```

**Review by nhudinh0309 — COMMENTED** [review]
Just had a small comment. Others look fine to me.

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts [hunk 14]
```diff
@@ -0,0 +1,257 @@
...
+  expect(await umbracoApi.document.doesNameExist(contentName)).toBeTruthy();
+  const contentData = await umbracoApi.document.getByName(contentName);
+  expect(contentData.variants[0].state).toBe(expectedState);
+});
+
+test('can publish content with a image media picker', async ({umbracoApi, umbracoUi}) => {
+  // Arrange
+  const expectedState = 'Published';
+  const dataType = await umbracoApi.dataType.getByName(dataTypeName);
+  const documentId = await umbracoApi.documentType.createDocumentTypeWithPropertyEditor(documentTypeName, dataTypeName, dataType.id, groupName);
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Content/ContentWithImageMediaPicker.spec.ts:47]: The variable "documentId" should be "documentTypeId"

**Pull request merged**